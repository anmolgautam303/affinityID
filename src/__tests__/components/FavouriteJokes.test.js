import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16'

import FavouriteJokes from '../../components/favouriteJokes';

configure({ adapter: new Adapter() });

describe('FavouriteJokes', () => {
    
    let props = { savedJokes: [] };
    let favouriteJokes = shallow(<FavouriteJokes {...props}/>);

    it('renders properly', () => {
        expect(FavouriteJokes).toMatchSnapshot();
    });

    it('should not render if prop `savedJokes` is an empty array', () => {
        expect(favouriteJokes.get(0)).toBeFalsy()
    });

    describe('when joke length is not zero', () => {
        beforeEach(() => {
          props = { savedJokes: [{ "id": "25vBR41iRNiXVzm_-IGxag", "value": "this is a test joke" }] };
          favouriteJokes = shallow(<FavouriteJokes {...props}/>);
        });

        it('div with class `container` renders properly', () => {
            expect(favouriteJokes.find('h4').text()).toEqual('Favourite jokes:');
        });

    });
});