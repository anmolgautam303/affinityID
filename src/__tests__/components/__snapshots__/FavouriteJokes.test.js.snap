// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavouriteJokes renders properly 1`] = `[Function]`;
